snap/snapcraft.yaml

```diff
@@ -0,0 +1,134 @@
+name: opensearch # you probably want to 'snapcraft register <name>'
+base: core20 # the base snap is the execution environment for this snap
+
+version: '2.2.0' # just for humans, typically '1.2+git' or '1.3.2'
+
+summary: 'OpenSearch: community-driven, Apache 2.0-licensed search and analytics suite.'
+description: |
+    OpenSearch is a community-driven, Apache 2.0-licensed open source search and 
+    analytics suite that makes it easy to ingest, search, visualize, and analyze data. 
+    Developers build with OpenSearch for use cases such as application search, 
+    log analytics, data observability, data ingestion, and more.
+
+grade: devel # must be 'stable' to release into candidate/stable channels
+
+confinement: strict # use 'strict' once you have the right plugs and slots
+
+
+assumes:
+  - command-chain
+  - snap-env
+  - snapd2.56
+
+
+system-usernames:
+  snap_daemon: shared
+
+
+plugs:
+  systemd-write:
+    interface: system-files
+    write:
+      - /etc/systemd/system/snap.opensearch.daemon.d
```

Comment:
this probably needs to be discussed with the security team

Reply:
Will do, and since it seems we can't use `sysctl` in strict confinement, it seems I'll have to also add the write permissions to the following list:
```
      - /proc/sys/vm/swappiness
      - /proc/sys/vm/max_map_count
      - /proc/sys/net/ipv4/tcp_retries2
```

Reply:
I removed this, the settings put there could be replaced with `ulimit` calls